test/utility.h

```diff
@@ -71,22 +71,25 @@ class TestVM : public testing::TestWithParam<std::string> {
     runtime_ = GetParam();
     if (runtime_ == "") {
       EXPECT_TRUE(false) << "runtime must not be empty";
-#if defined(WASM_V8)
+#if defined(PROXY_WASM_HAS_RUNTIME_V8)
     } else if (runtime_ == "v8") {
       vm_ = proxy_wasm::createV8Vm();
 #endif
-#if defined(WASM_WAVM)
+#if defined(PROXY_WASM_HAS_RUNTIME_WAVM)
     } else if (runtime_ == "wavm") {
       vm_ = proxy_wasm::createWavmVm();
 #endif
-#if defined(WASM_WASMTIME)
+#if defined(PROXY_WASM_HAS_RUNTIME_WASMTIME)
     } else if (runtime_ == "wasmtime") {
       vm_ = proxy_wasm::createWasmtimeVm();
 #endif
-#if defined(WASM_WAMR)
+#if defined(PROXY_WASM_HAS_RUNTIME_WAMR)
     } else if (runtime_ == "wamr") {
       vm_ = proxy_wasm::createWamrVm();
 #endif
+    } else {
+      EXPECT_TRUE(false) << "compiled without support for the requested \"" << runtime_
```

Comment:
nit, better use `FAIL()` ?

Reply:
`FAIL()` cannot be used in a constructor, since it returns. It seems that you've run into the same issue few lines above, where `EXPECT_TRUE(false)` is also used.

Reply:
ah OK, thanks!